wandb/sdk/internal/internal_api.py

```diff
@@ -3485,6 +3490,22 @@ def link_artifact(
                     }
             """
 
+        org_entity = ""
+        if is_artifact_registry_project(project):
+            org_fields = self.server_organization_introspection()
+            if "orgEntity" not in org_fields:
+                raise ValueError(
+                    """Fetching Registry artifacts without inputting an organization is unavailable for your server version.
+                    Please upgrade your server"""
+                )
+            # Fetch the org_entity and org name that we are expecting. Note: this is making assumption that
+            org_entity, org_name = self.fetch_org_entity_from_entity(entity)
+            if organization:
+                if organization != org_name and organization != org_entity:
+                    raise ValueError(
+                        f"Input wrong organization: {organization} for registry: {project}/{portfolio_name}"
+                    )
```

Comment:
suggestion: might be good to reduce the nesting of if-statements here.

among other things, how do you feel about pulling out a (private) helper function -- like `self._resolve_org_entity(...)` -- for the block inside `if is_artifact_registry_project():`?  E.g. something like.:
```python3
        if is_artifact_registry_project(project):
            org_entity = self._resolve_org_entity(organization, entity, project, portfolio_name)
        else:
            org_entity = ""
```